maths/allocation_number.py

```diff
@@ -35,7 +34,7 @@ def allocation_num(number_of_bytes: int, partitions: int) -> list[str]:
     bytes_per_partition = number_of_bytes // partitions
     allocation_list = []
     for i in range(partitions):
-        start_bytes = i * bytes_per_partition + 1
+        start_bytes = 0 if i * bytes_per_partition == 0 else i * bytes_per_partition + 1
```

Comment:
This makes the first block larger than all other blocks and allocates one block too many.